lib/ipinfo-rails.rb

```diff
@@ -6,28 +6,15 @@
 class IPinfoMiddleware
     def initialize(app, cache_options = {})
         @app = app
-        @token = cache_options.fetch(:token, nil)
+        @token = cache_options.fetch(:token, nil) || ENV['IPINFO_TOKEN']
         @ipinfo = IPinfo.create(@token, cache_options)
         @filter = cache_options.fetch(:filter, nil)
     end
 
     def call(env)
-        env['called'] = 'yes'
         request = Rack::Request.new(env)
-
-        filtered = if @filter.nil?
-                       is_bot(request)
-                   else
-                       @filter.call(request)
-                   end
-
-        if filtered
-            env['ipinfo'] = nil
-        else
-            ip = request.ip
-            env['ipinfo'] = @ipinfo.details(ip)
-        end
-
+        filtered = @filter.nil? ? is_bot(request) : @filter.call(request)
+        env['ipinfo'] = filtered ? nil : @ipinfo.details(request.env["HTTP_X_FORWARDED_FOR"] || request.ip)
```

Comment:
The x-forwarded-for support here looks irrelevant to Heroku, right? It should be made as a separate commit if so, and also made to align with the requirements set in https://github.com/ipinfo/rails/issues/6 which require a handler to be called to get the IP, with the default doing what's being done here.

Reply:
@UmanShahzad `x-forwarded-for` is a requirement of Heroku. https://devcenter.heroku.com/articles/http-routing#heroku-headers. Cloudflare, Google Cloud also support it. My assumption is most services are backed up by a frontend load balancer so this would be the default.

We can get to https://github.com/ipinfo/rails/issues/6 at a later stage as I am focussed on getting this live.

Reply:
Alright makes sense - I should've known Heroku would intercept all requests at their own load balancer.

But one more question before we can safely merge: can't this return a comma-separated list of IPs? We would need to pick the first one in the list in case more than 1 IP is tacked on, no?

https://stackoverflow.com/questions/18264304/get-clients-real-ip-address-on-heroku

Reply:
Alright I suggest reading the discussion on https://github.com/ipinfo/node-express/issues/5 before we merge this.

Reply:
I do not have any strong opinion around `x-forwarded-for` or custom header support as I haven't run in such cases or required them. Would be good to support them.

The changes suggested in this PR have been tested on Heroku by a customer.
https://github.com/ipinfo/rails/tree/heroku
https://secure.helpscout.net/conversation/1314160026/80743/

Reply:
@TroySK The issue is, as the above 2 links imply, this can possibly break things for some non-heroku users. We would need the approach of allowing an override, and then allowing heroku users to choose a specific override.

Reply:
@UmanShahzad I am not sure what the right solution is based on the conversations and I need more inputs as to what the convention needs to be across the different SDKs.